app/src/main/java/app/wefridge/wefridge/EditFragment.kt

```diff
@@ -17,39 +51,548 @@ import app.wefridge.wefridge.placeholder.PlaceholderContent
+            registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted: Boolean ->
+                if (isGranted) {
+                    getCurrentLocation()
+                } else {
+                    Log.d("EditFragment", "Request for location access denied.")
+                }
+            }
     }
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
+        // this line of code was partially inspired by https://stackoverflow.com/questions/11741270/android-sharedpreferences-in-fragment
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(requireActivity())
+
         // Inflate the layout for this fragment
         _binding = FragmentEditBinding.inflate(inflater, container, false)
 
         return binding.root
     }
 
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        adaptUIToModel()
+        hideDatePicker()
+        setUpUnitDropdown()
+        setLocationPickerActivation()
+        setUpOnClickListenersForFormComponents()
+        setUpSaveMechanism()
+        setUpOnDateChangedListenerForDatePicker()
+        setUpOnFocusChangeListenerForAddressInputEditText()
+        location = model?.location
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        if (!ADD_ITEM_MODE) {  // **new** Items should not be saved automatically, i. e. onDestroy
+            // TODO: put the following condition into a separate function
+            if (location == null && model?.isShared == true) {
+                model?.location = null
+                model?.geohash = null
+                model?.isShared = false
+                displayAlertOnSaveSharedItemWithoutLocation()
+            } else {
+                setModelLocationAttribute()
+                setModelGeohashAttribute()
+            }
+            setModelContactNameAttribute()
+            setModelContactEmailAttribute()
+
+            // TODO: put the following condition into a separate function
+            if ((model?.contactEmail == null || model?.contactEmail == "") && model?.isShared == true) {
+                model?.isShared = false
+                displayAlertOnSaveSharedItemWithoutContactEmail()
+            }
+
+            val itemController: ItemControllerInterface = ItemController()
+            itemController.saveItem(model!!, { /* do nothing on success */ }, { displayAlertOnSaveItemFailed() })
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun setUpOnClickListenersForFormComponents() {
+
+        unitDropdownMenu.setOnMenuItemClickListener { menuItem ->
+            unit_dropdown.editText?.setText(menuItem.title)
+            setModelUnitAttribute()
+            true
+        }
+
+        unit_dropdown.editText?.setOnClickListener() {
```

Comment:
Is that log still interesting?

Reply:
No, thank you for noticing that. I have removed it and will commit the change soon.